packages/trader/src/AppV2/Hooks/useGuideStates.ts

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+type GuideFlags = 'should_run_trade_page_guide' | 'should_run_market_selector_guide';
+
+type GuideStates = Record<GuideFlags, boolean>;
+
+let sharedGuideStates: GuideStates = {
+    should_run_trade_page_guide: false,
+    should_run_market_selector_guide: false,
+};
+
+const listeners = new Set<() => void>();
+
+const useGuideStates = () => {
+    const [, setUpdate] = React.useState({});
+
+    React.useEffect(() => {
+        const update = () => setUpdate({});
```

Comment:
Okay, so you are creating a Set with callbacks, that will trigger re-rendering. On each callback call you are changing sharedGuideStates, correct?
Code below won't work?
const useGuideStates = () => {
   const [guideStates, setGuideStates] = React.useState({should_run_trade_page_guide: false,
    should_run_market_selector_guide: false,});
    
   const setGuideState = (flag: GuideFlags, value: boolean) => setGuideStates(prev => {...prev,[flag]: value });
   
   return {
        guideStates,
        setGuideState,
    };
}

Reply:
the reason for using a set and listener is to keep the same state across all components whenever this hook is called. Otherwise, a brand new state is created whenever the hook is called. I can do much easier implementation if i had just used a store but we were trying to move away from using the store so had to find a different way.